dvc/command/gc.py

```diff
@@ -12,17 +12,18 @@
 
 class CmdGC(CmdBase):
     def run(self):
-        msg = "This will remove all cache except items used in "
-
-        msg += "the working tree"
-        if self.args.all_commits:
-            msg += " and all git commits"
-        elif self.args.all_branches and self.args.all_tags:
-            msg += " and all git branches and tags"
-        elif self.args.all_branches:
-            msg += " and all git branches"
-        elif self.args.all_tags:
-            msg += " and all git tags"
+        msg = (
+            "This will remove all cache except items used in the working tree"
+            "{tags}{history}{branches}{history}"
+        ).format(
+            tags="" if self.args.remove_all_tags else "and all git tags",
+            branches=""
+            if self.args.remove_all_branches
+            else "and all git branches",
+            history=""
+            if self.args.remove_all_history
+            else "and their history",
```

Comment:
This makes the phrase weird - many ands. Maybe just list things instead:
```
This will remove all cache not referenced from:
- the working tree
- all branch heads
...
```
? Will make it more glanceable, the changing items will be always in the same place, any user already familiar with gc, will be simply skipping the entering phrase without loose of any meaning.